OpenRA.Mods.Common/Graphics/TilesetSpecificSpriteSequence.cs

```diff
@@ -32,20 +32,56 @@ public class TilesetSpecificSpriteSequence : DefaultSpriteSequence
 		[Desc("Dictionary of <tileset name>: filename to override the Filename key.")]
 		static readonly SpriteSequenceField<Dictionary<string, string>> TilesetFilenames = new SpriteSequenceField<Dictionary<string, string>>(nameof(TilesetFilenames), null);
 
-		public TilesetSpecificSpriteSequence(ModData modData, string tileset, SpriteCache cache, ISpriteSequenceLoader loader, string image, string sequence, MiniYaml data, MiniYaml defaults)
-			: base(modData, tileset, cache, loader, image, sequence, data, defaults) { }
+		public TilesetSpecificSpriteSequence(SpriteCache cache, ISpriteSequenceLoader loader, string image, string sequence, MiniYaml data, MiniYaml defaults)
+			: base(cache, loader, image, sequence, data, defaults) { }
 
-		protected override string GetSpriteFilename(ModData modData, string tileset, string image, string sequence, MiniYaml data, MiniYaml defaults)
+		protected override IEnumerable<(string Filename, int[] LoadFrames, int[] Frames, MiniYamlNode.SourceLocation Location)> ParseFilenames(ModData modData, string tileset, int[] frames, MiniYaml data, MiniYaml defaults)
 		{
 			var node = data.Nodes.FirstOrDefault(n => n.Key == TilesetFilenames.Key) ?? defaults.Nodes.FirstOrDefault(n => n.Key == TilesetFilenames.Key);
 			if (node != null)
 			{
 				var tilesetNode = node.Value.Nodes.FirstOrDefault(n => n.Key == tileset);
 				if (tilesetNode != null)
-					return tilesetNode.Value.Value;
+				{
+					// Only request the subset of frames that we actually need
```

Comment:
These tileset sequence implementations for ParseFilenames/ParseCombineFilenames occur twice. Is it worth having it shared somewhere? Maybe a protected helper in the base class that impls can choose to employ in this override?